llmfoundry/models/mpt/modeling_mpt.py

```diff
@@ -995,6 +987,16 @@ def _reorder_cache(
         return reordered_past
 
 
+def _fsdp_wrap_fn(self: Union[MPTModel, MPTForCausalLM],
+                  module: nn.Module) -> bool:
+    # FSDP Wrap function for MPT Models
+    return isinstance(module, MPTBlock)
+
+
+MPTModel.fsdp_wrap_fn = _fsdp_wrap_fn
+MPTForCausalLM.fsdp_wrap_fn = _fsdp_wrap_fn
```

Comment:
nit: Can u put fn above class?

Also dont monkeypatch, call helper fn directly from `fsdp_wrap_fn`?

Reply:
refac'd so that fn is above class
Note: pyright complains that `self` types (ie `Union[MPTModel, MPTForCausalLM]`) are not defined yet, so I had to add the `# type: ignore[no-untyped-def]`

Reply:
I sitll dont really like. 

Also youc an use from future import annotations to type self. Wrap name of type in quotes eg ('MPTModel') and it will work